snuba/query/processors/apdex_processor.py

```diff
@@ -0,0 +1,74 @@
+from snuba.query.conditions import (
+    binary_condition,
+    BooleanFunctions,
+    ConditionFunctions,
+)
+from snuba.query.expressions import (
+    Expression,
+    FunctionCall,
+    Literal,
+)
+from snuba.query.dsl import div, multiply, plus
+from snuba.query.query import Query
+from snuba.query.query_processor import QueryProcessor
+from snuba.request.request_settings import RequestSettings
+
+
+class ApdexProcessor(QueryProcessor):
+    """
+    Resolves the apdex function call into
+    countIf({col} <= {satisfied}) + (countIf(({col} > {satisfied}) AND ({col} <= {tolerated})) / 2)) / count()
+    according to the definition of apdex index: https://en.wikipedia.org/wiki/Apdex
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_functions(exp: Expression) -> Expression:
+            if isinstance(exp, FunctionCall) and exp.function_name == "apdex":
+                assert len(exp.parameters) == 2
+                column = exp.parameters[0]
+                satisfied = exp.parameters[1]
+                tolerated = multiply(satisfied, Literal(None, 4))
+
+                return div(
+                    plus(
+                        FunctionCall(
```

Comment:
Is threre a reason why some of these (div, plus and the other arithmetic funtions) are defined as functions and countIf isn't?

Reply:
I wanted to keep the DSL small, so I ported only the arithmetic functions that are likely to be used often. I did not yet decide what to do with count and other aggregate functions, we  may add them to the DSL if they turn out to be useful. I did not use them outside of tests so far.